GCMessengerSDKSample/app/src/main/res/layout/fragment_chat_form.xml

```diff
@@ -18,44 +18,58 @@
     <androidx.appcompat.widget.AppCompatTextView
         android:layout_width="wrap_content"
         android:layout_height="wrap_content"
-        android:text="Account Details"
+        android:text="@string/title_details"
         android:textColor="@color/black"
         android:textSize="22sp"
         android:layout_margin="10dp"/>
 
-    <androidx.appcompat.widget.AppCompatEditText
-        android:id="@+id/deploymentIdEditText"
+    <ScrollView
         android:layout_width="match_parent"
-        android:layout_height="wrap_content"
-        android:hint= "@string/quick_start_account_form_account_name" />
+        android:layout_height="0dp"
+        android:layout_weight="1">
 
-    <androidx.appcompat.widget.AppCompatEditText
-        android:id="@+id/domainNameEditText"
-        android:layout_width="match_parent"
-        android:layout_height="wrap_content"
-        android:hint="@string/quick_start_account_form_knowledge_base" />
+        <LinearLayout
+            android:id="@+id/account_fields"
+            android:layout_width="match_parent"
+            android:layout_height="wrap_content"
+            android:orientation="vertical"
+            android:visibility="visible">
 
-    <androidx.appcompat.widget.AppCompatEditText
-        android:id="@+id/customAttributesEditText"
-        android:layout_width="match_parent"
-        android:layout_height="wrap_content"
-        android:hint="@string/quick_start_account_form_custom_attributes" />
+            <androidx.appcompat.widget.AppCompatEditText
+                android:id="@+id/deploymentIdEditText"
+                android:layout_width="match_parent"
+                android:layout_height="wrap_content"
+                android:hint="@string/quick_start_account_form_account_name" />
 
-    <androidx.appcompat.widget.SwitchCompat
-        android:id="@+id/loggingSwitch"
-        android:layout_width="match_parent"
-        android:layout_height="wrap_content"
-        android:enabled="true"
-        android:padding="6sp"
-        android:text="@string/quick_start_account_form_logging"
-        android:textSize="16sp"
-        android:layout_marginTop="6dp"/>
+            <androidx.appcompat.widget.AppCompatEditText
+                android:id="@+id/domainNameEditText"
+                android:layout_width="match_parent"
+                android:layout_height="wrap_content"
+                android:hint="@string/quick_start_account_form_knowledge_base" />
 
+            <androidx.appcompat.widget.AppCompatEditText
+                android:id="@+id/customAttributesEditText"
+                android:layout_width="match_parent"
+                android:layout_height="wrap_content"
```

Comment:
I would try here to set the following layoutParams:
```suggestion
                android:layout_width="match_parent"
        android:layout_height="0dp"
        android:layout_weight="1"
```
Maybe then you can remove the ScrollView (and set its layoutParams right into the LinearLayout)

Reply:
1. Tried the suggestion and it doesn't work as expected because the Text Fields grows to fill the empty space even when it's empty.
2. I'm comparing the layout with the similar one in https://bitbucket.org/inindca/mobiledx-android-internal-app/src/master/app/src/main/res/layout/account_form.xml

I think it worth unifying them.
What do you think?

My next question is why do we have two apps that do similar things?
- mobiledx-android-internal-app
- mobiledx-samples-android
Is it because `mobiledx-android-internal-app` has internal things not exposed to customer?

Reply:
Account form is legacy. Once we have time, we will remove it. So don't have to unify them.

Sample app intentionally is for customers to see the usage of SDK. Just like you phrased in the README.
In practice we use this to run the automated end-to-end tests. That makes this app a honorary test application.

Internal app intentionally is for development. It depends on the code of the SDK which means we don't have to make artifacts out of them, when we want to run. My opinion is that this should be the test application instead of the Sample app. Maybe once we will make this change. But in any case we need two different application.

Reply:
Thank you for explanation! 

I think `mobiledx-android-internal-app` also deserves own README file.

I also think if two apps have some common code, instead of supporting it in two places we can look on sharing the same between those two.